pkg/controller/certificates/sync.go

```diff
@@ -216,6 +216,12 @@ func (c *Controller) certificateMatchesSpec(crt *v1alpha1.Certificate, key crypt
 		errs = append(errs, fmt.Sprintf("DNS names on TLS certificate not up to date: %q", cert.DNSNames))
 	}
 
+	// validate the ip addresses are correct
+	expectedIPAddresses := pki.IPAddressesNameForCertificate(crt)
```

Comment:
Given we don't modify `crt.spec.ipAddresses` as part of this check, can we remove the `IPAddressesNameForCertificate` function altogether in favour of just referencing this field directly in this area of code?

This is the only place this function is actually called outside of the PKI package 😄